Fix: install Terraform and apply the support for a provider in one jump

zzxwill opened the pull request:
- Fix issue: addon name should not include forward-slash "/" cloud-resource   fix #2588
- Install terraform and the provider in one jump. Instead of installing `terraform`
and `terraform-xxx` in two steps
- Terraform ComponenetDefinition won't be applied at once even though not all of them are needed


### Description of your changes

<!--

Briefly describe what this pull request does. We love pull requests that resolve an open KubeVela issue. If yours does, you
can uncomment the below line to indicate which issue your PR fixes, for example
"Fixes #500":

-->

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

<!--
Before reviewers can be confident in the correctness of this pull request, it
needs to tested and shown to be correct. Briefly describe the testing that has
already been done or which is planned for this change.
-->


### Special notes for your reviewer

<!--

Be sure to direct your reviewers'
attention to anything that needs special consideration.

-->

zzxwill commented:
Please hold on to this PR until https://github.com/oam-dev/kubevela/pull/2723 is merged.

wonderflow reviewed (commented):
why fluxcd related in this PR？

zzxwill commented:
> why fluxcd related in this PR？

My local master isn't the latest, at a moment, I rebased master to this branch. But this should not happen, I guess. 
I have rebased the latest, all fulxcd related pieces are cleaned up.

zzxwill commented:
- addon `terraform-xxx` can be successfully enabled.
```
➜  kubevela git:(deprecate-terraform) ✗ vela addon enable terraform-alibaba ALICLOUD_ACCESS_KEY=LTAI5tCjMFZQLNkgLnRyWvay ALICLOUD_SECRET_KEY=5IcTklyhUZVw75rmu50KxeYb5krAdA ALICLOUD_REGION=cn-hongkong
Application terraform-alibaba is in phase:...
```

And all related kinds of stuff like `secret` and `Provider` are successfully created. An VPC application is successfully provisioned.

- addon `terraform` doesn't show up in `vela addon list`
```
➜  kubevela git:(deprecate-terraform) go run references/cmd/cli/main.go addon list
NAME               	DESCRIPTION                                                                  	STATUS
fluxcd             	Flux is a set of continuous and progressive delivery solutions for Kubernetes	installed
istio              	istio Controller is a Kubernetes Controller for manage traffic.              	uninstalled
kruise             	Kruise is a Kubernetes extended suite for application automations            	uninstalled
observability      	An out of the box solution for KubeVela observability                        	uninstalled
ocm-cluster-manager	ocm-cluster-manager can deploy an OCM hub cluster environment.               	uninstalled
terraform-alibaba  	Kubernetes Terraform Controller for Alibaba Cloud                            	installed
terraform-aws      	Kubernetes Terraform Controller for AWS                                      	uninstalled
terraform-azure    	Kubernetes Terraform Controller for Azure                                    	uninstalled
```

chivalryq commented:
Should we keep terrafrom away from user all the time? Is that neccesarry? Because now we have "decline the enabling addon when dependencies not satisfied" in apiserver

zzxwill commented:
> Should we keep terrafrom away from user all the time? 

No. Use doesn't need `terraform` at all. Addon `terraform` will be automatically installed by addon `terraform-xxxx`。

> Is that neccesarry? 

Are there any scenario that could not cover after this change?

> Because now we have "decline the enabling addon when dependencies not satisfied" in apiserver

What does it mean?

chivalryq commented:
After talking offline, we decide to add `invisible` field in addon's format , default to be false.

This flag means this addon should be enabled implicitly. When some addon depends this addon, it'll be enabled.

In both UI and CLI, invisible addon won't be listed.

Now this is a transitional implementation.

zzxwill commented:
> After talking offline, we decide to add `invisible` field in addon's format , default to be false.
> 
> This flag means this addon should be enabled implicitly. When some addon depends this addon, it'll be enabled. This addon shoud **NOT** need arguments.
> 
> In both UI and CLI, invisible addon won't be listed.
> 
> Now this is a transitional implementation.

Great:) Let's refactor these lines later https://github.com/oam-dev/kubevela/pull/2718/files#diff-5c1ce01bb88762ec6a5abb96531e8ac946cff33a1fa7ff2e033131f8dcf0e2b8R195-R198.
